testing/test_tmpdir.py

```diff
@@ -139,6 +140,24 @@ def test_1(tmp_path):
             # Check the base dir itself is gone
             assert len(list(base_dir)) == 0
 
+    # issue #10502
+    def test_delete_dir_when_skipped_from_fixture(self, pytester: Pytester) -> None:
+        p = pytester.makepyfile(
+            """
+            import pytest
+
+            @pytest.fixture
+            def fixt(tmp_path):
+                pytest.skip()
+
+            def test_fixt(fixt):
+                pass
+        """
+        )
+
+        reprec = pytester.inline_run(p)
+        assert reprec.ret == ExitCode.OK
```

Comment:
There should probably also be an assertion here which makes sure the files are still around (or not around, depending on policy) - see the earlier comment above.

Reply:
Added some tests.

[e4cf09d](https://github.com/pytest-dev/pytest/pull/10517/commits/e4cf09d8cb03775a1b8598bc3e15898ab2d18b89)